app/Http/Frontend/Controllers/CartController.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\Http\Frontend\Controllers;
+
+use Illuminate\Http\Request;
+use App\Http\Frontend\Requests;
+use Cart;
+use DB;
+use App\Models\Product;
+
+class CartController extends Controller
+{
+    /**
+     * Display a listing of the new products.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $content = Cart::content();
+        $total = Cart::total();
+        return view('frontend.cart.index', compact('content', 'total'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @param int $id id product
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function addProduct($id)
+    {
+        $product = Product::findOrFail($id);
+        Cart::add(['id' => $id,
```

Comment:
I think it's not good to add an array to Cart, why don't you add a Product Eloquent Object ?